src/parser/transform/transform_ddl.cpp

```diff
@@ -87,9 +92,98 @@ std::unique_ptr<Statement> Transformer::transformCreateRdfGraphClause(
     return std::make_unique<CreateTable>(std::move(createTableInfo));
 }
 
-std::unique_ptr<Statement> Transformer::transformDropTable(CypherParser::KU_DropTableContext& ctx) {
-    auto tableName = transformSchemaName(*ctx.oC_SchemaName());
-    return std::make_unique<Drop>(tableName);
+std::unique_ptr<Statement> Transformer::transformCreateSequence(
+    CypherParser::KU_CreateSequenceContext& ctx) {
+    auto sequenceName = transformSchemaName(*ctx.oC_SchemaName());
+    auto createSequenceInfo = CreateSequenceInfo(sequenceName);
+    std::unordered_set<SequenceInfoType> applied;
+    bool defaultStart = true;
+    bool defaultMin = true;
+    bool defaultMax = true;
+    for (auto seqOption : ctx.kU_SequenceOptions()) {
+        SequenceInfoType type;
+        std::string typeString;
+        CypherParser::OC_IntegerLiteralContext* valCtx = nullptr;
+        int64_t* valOption = nullptr;
+        std::string rawVal = "";
+        if (seqOption->kU_StartWith()) {
+            type = SequenceInfoType::START;
+            typeString = "START";
+            rawVal = seqOption->kU_StartWith()->MINUS() ? "-" : "";
+            valCtx = seqOption->kU_StartWith()->oC_IntegerLiteral();
+            valOption = &createSequenceInfo.startWith;
+            defaultStart = false;
+        } else if (seqOption->kU_IncrementBy()) {
+            type = SequenceInfoType::INCREMENT;
+            typeString = "INCREMENT";
+            rawVal = seqOption->kU_IncrementBy()->MINUS() ? "-" : "";
+            valCtx = seqOption->kU_IncrementBy()->oC_IntegerLiteral();
+            valOption = &createSequenceInfo.increment;
+        } else if (seqOption->kU_MinValue()) {
+            type = SequenceInfoType::MINVALUE;
+            typeString = "MINVALUE";
+            if (!seqOption->kU_MinValue()->NO()) {
+                rawVal = seqOption->kU_MinValue()->MINUS() ? "-" : "";
+                valCtx = seqOption->kU_MinValue()->oC_IntegerLiteral();
+                valOption = &createSequenceInfo.minValue;
+                defaultMin = false;
+            }
+        } else if (seqOption->kU_MaxValue()) {
+            type = SequenceInfoType::MAXVALUE;
+            typeString = "MAXVALUE";
+            if (!seqOption->kU_MaxValue()->NO()) {
+                rawVal = seqOption->kU_MaxValue()->MINUS() ? "-" : "";
+                valCtx = seqOption->kU_MaxValue()->oC_IntegerLiteral();
+                valOption = &createSequenceInfo.maxValue;
+                defaultMax = false;
+            }
+        } else { // seqOption->kU_Cycle()
+            type = SequenceInfoType::CYCLE;
+            typeString = "CYCLE";
+            if (!seqOption->kU_Cycle()->NO()) {
+                createSequenceInfo.cycle = true;
+            }
+        }
+        if (applied.find(type) != applied.end()) {
```

Comment:
All these checks from l147 - l164 should be moved into binder. The parser should only check for grammar correctness. Whether integer are out of bound or non-zero are semantical checks that should be done in binder.

Reply:
i think l147-l150 makes sense here since I think checking duplication clauses is part of grammar correctness. Going to move the integer casts to binder.